x-pack/plugins/security/server/authorization/types.ts

```diff
@@ -71,6 +71,6 @@ export interface CheckPrivilegesPayload {
   kibana?: string | string[];
   elasticsearch?: {
     cluster: string[];
-    index: Record<string, string[]>;
+    index?: Record<string, string[]>;
```

Comment:
Hmm, I thought [`check_privileges.ts`](https://github.com/elastic/kibana/blob/8a3bda4cfd0016e255465ccc2a601c38a1cd7ef7/x-pack/plugins/security/server/authorization/check_privileges.ts#L87-L97) ensured that we'd never return `undefined` for this property. Do you recall why this change was required?

Reply:
This change is for CheckPrivilegesPayload, not CheckPrivilegesResponse. You can create a checkPrivileges request without setting elasticsearch.index.

Reply:
🙈 thanks, disregard!

Reply:
I ended up removing this change and just providing a value of `{}` if index is not provided.